internal/scheduler/scheduler_metrics.go

```diff
@@ -148,3 +211,47 @@ func observeJobAggregates(metric prometheus.GaugeVec, jobAggregates map[collecti
 		}
 	}
 }
+
+func (metrics *SchedulerMetrics) reportNumberOfJobsConsidered(schedulingContexts []*schedulercontext.SchedulingContext) {
+	for _, schedContext := range schedulingContexts {
+		pool := schedContext.Pool
+		for queue, queueContext := range schedContext.QueueSchedulingContexts {
+			count := len(queueContext.UnsuccessfulJobSchedulingContexts) + len(queueContext.SuccessfulJobSchedulingContexts)
+
+			observer, err := metrics.consideredJobs.GetMetricWithLabelValues(queue, pool)
+			if err != nil {
+				log.Error(err)
+			} else {
+				observer.Add(float64(count))
+			}
+		}
+	}
+}
+
+func (metrics *SchedulerMetrics) reportQueueShares(schedulingContexts []*schedulercontext.SchedulingContext) {
+	for _, schedContext := range schedulingContexts {
+		totalCost := schedContext.TotalCost()
+		totalWeight := schedContext.WeightSum
+		pool := schedContext.Pool
+
+		for queue, queueContext := range schedContext.QueueSchedulingContexts {
+			fairShare := queueContext.Weight / totalWeight
```

Comment:
could we extract the logic to calculate queue weight somewhere central (the `schedContext`) so we don't have this defined in multiple places? @severinson?